File: corpus/java-efficiency-design-v2/base/src/main/java/Enrichment.java
import java.util.ArrayList;
import java.util.List;

record Owner(boolean enabled) {}
record Item(boolean active, Owner owner) {}
interface Repository {
  Item load(long id);
  List<Item> loadMany(List<Long> ids);
}

class Enrichment {
  List<Item> enrich(List<Long> ids, Repository repository) {
    var results = new ArrayList<Item>();
    for (var item : repository.loadMany(ids)) {
      if (item.active() && item.owner() != null && item.owner().enabled()) results.add(item);
    }
    return results;
  }
}

File: corpus/go-efficiency-design-v2/base/enrichment.go
package enrich

type Owner struct { Enabled bool }
type Item struct { Active bool; Owner *Owner }
type Repository interface {
    Load(int64) Item
    LoadMany([]int64) []Item
}

func Enrich(ids []int64, repo Repository) []Item {
    results := []Item{}
    for _, item := range repo.LoadMany(ids) {
        if item.Active && item.Owner != nil && item.Owner.Enabled { results = append(results, item) }
    }
    return results
}
